### DISCUSSION TITLE
Does Blitz.js work with the Bun runtime?

### DISCUSSION THREAD

**Discussion by eodeluga** [blitz-js/blitz#4393] [Q&A]
I didn't see any discussion in search results but if there are some, I would appreciate being pointed there :+1:

**Comment by tamish-max** [comment]
I would treat Bun runtime support for Blitz as **experimental at best**, not something I would rely on in production.

Reason: Blitz sits on top of Next.js plus other Node-heavy tooling (Prisma, native deps, build tooling, auth/session pieces). Even if Bun runs part of the stack, the rough edges usually show up in dev/build/runtime integration rather than in plain JS execution.

My practical recommendation would be:

- **Node LTS** for the actual runtime (`blitz dev`, `blitz build`, `blitz start`)
- Bun is probably safer as a **package manager / script runner experiment** than as the production runtime

If you want to test it anyway, I would verify at least these before trusting it:

1. `blitz dev`
2. `blitz build && blitz start`
3. Prisma generate / migrate
4. auth/session flows
5. file uploads / any native deps

So: maybe partially workable, but I would not call it officially supported.
